### DISCUSSION TITLE
Annotated-style Form definitions breaking when setting default value and having multiple routes pointing to the same function

### DISCUSSION THREAD

**Discussion by harpaj** [fastapi/fastapi#9306] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, Form

app = FastAPI()


@app.post("/working")
@app.post("/working_too")
async def working(sample: str | None = Form(default=None)) -> list[str | None]:
    return [sample]


@app.post("/working_also")
async def working_also(sample: Annotated[str | None, Form()] = None) -> list[str | None]:
    return [sample]


@app.post("/broken")
@app.post("/broken_too")
async def broken(sample: Annotated[str | None, Form()] = None) -> list[str | None]:
    return [sample]
```

### Description

The two functions `working` and `broken` should be equivalent, valid ways of defining a Form according to the documentation.
However, in the case where two routes are pointing to the same function and a default value is defined, it breaks during start-up with 

```
  File "/home/trustyou.local/johannes.huber/Documents/fix_dashboards/venv/lib/python3.10/site-packages/fastapi/dependencies/utils.py", line 387, in analyze_param
    assert field_info.default is Undefined or field_info.default is Required, (
AssertionError: `Form` default value cannot be set in `Annotated` for 'sample'. Set the default value with `=` instead.
```

It seems to happen only with that specific combination, it's both fine when setting no default values and when only using one route to the function.

### Operating System

Linux

### FastAPI Version

0.95.0

### Python Version

Python 3.10.8

**Comment by sharonyogev** [comment, marked as answer]
This should solve it: https://github.com/tiangolo/fastapi/pull/9315

**Reply by harpaj** [reply]
Thanks a lot! Hoping it will be merged and released soon.

**Reply by tiangolo** [reply]
This should be fixed in https://github.com/tiangolo/fastapi/pull/9315

Just released in FastAPI 0.95.1 :tada:
